deltacat/aws/constants.py

```diff
@@ -1,10 +1,13 @@
 from typing import List, Set
 
+from botocore.exceptions import NoCredentialsError, ReadTimeoutError
+
 from deltacat.utils.common import env_integer, env_string
 
 DAFT_MAX_S3_CONNECTIONS_PER_FILE = env_integer("DAFT_MAX_S3_CONNECTIONS_PER_FILE", 8)
 BOTO_MAX_RETRIES = env_integer("BOTO_MAX_RETRIES", 5)
 TIMEOUT_ERROR_CODES: List[str] = ["ReadTimeoutError", "ConnectTimeoutError"]
 RETRYABLE_PUT_OBJECT_ERROR_CODES: Set[str] = {"Throttling", "SlowDown"}
+RETRYABLE_TRANSIENT_ERRORS = (NoCredentialsError, ReadTimeoutError, ConnectionError)
```

Comment:
Should we stick with error codes instead of using the errors directly?

Reply:
boto won't capture every error as a ClientError with error codes supplied, unfortunately. There was one instance where `s3.put_object` threw a SSLError as an example.